AxisCommunications/docker-acap
# TLS cert upload for rootless

A first rough draft of using fast cgi for TLS upload.
Still needs a cleanup, correct permissions and better handling of when certs are available vs. not available

How to test :
- use 11.9 fw 
- build and install the Docker ACAP
- start with use TLS False (ACAP still shuts down if certs are not available when use TLS is True)
- create certs on host as usual but use curl to upload (see commands below)
- with all three certs uploaded it should be possible to restart the ACAP with use TLS set to True (but not yet verified by me)

"Supported" curl commands
GET: curl --anyauth --user root:pass -X GET "http://$DEVICE_IP/local/dockerdwrapper/tls_reciever.cgi?name=Axis"
> Accepted but doesn't do anything, my thought was that it could be used for info to user
POST: curl --anyauth --user root:pass -F uploadedFile=@<file-to-upload> -X POST "http://$DEVICE_IP/local/dockerdwrapper/tls_reciever.cgi?file_name=<cert-file-name>"
> Uploads a cert file,  cert-file-name can be ca.pem, server-key.pem or server-cert.pem
DELETE: curl --anyauth --user root:pass -X DELETE "http://$DEVICE_IP/local/dockerdwrapper/tls_reciever.cgi?file_name=<cert-file-name>"
> Deletes a cert file,  cert-file-name can be ca.pem, server-key.pem or server-cert.pem
